src/sonic-host-services/scripts/hostcfgd

```diff
@@ -680,125 +866,6 @@ class HostConfigDaemon:
-        start_cmds = []
-        for feature_name in feature_names:
-            for suffix in feature_suffixes:
-                start_cmds.append("sudo systemctl unmask {}.{}".format(feature_name, suffix))
-            # If feature has timer associated with it, start/enable corresponding systemd .timer unit
-            # otherwise, start/enable corresponding systemd .service unit
-            start_cmds.append("sudo systemctl enable {}.{}".format(feature_name, feature_suffixes[-1]))
-            start_cmds.append("sudo systemctl start {}.{}".format(feature_name, feature_suffixes[-1]))
-            for cmd in start_cmds:
-                syslog.syslog(syslog.LOG_INFO, "Running cmd: '{}'".format(cmd))
-                try:
-                    subprocess.check_call(cmd, shell=True)
-                except subprocess.CalledProcessError as err:
-                    syslog.syslog(syslog.LOG_ERR, "'{}' failed. RC: {}, output: {}"
-                                  .format(err.cmd, err.returncode, err.output))
-                    syslog.syslog(syslog.LOG_ERR, "Feature '{}.{}' failed to be  enabled and started"
-                                  .format(feature_name, feature_suffixes[-1]))
-                    return
-
-    def disable_feature(self, feature_names, feature_suffixes):
-        stop_cmds = []
-        for feature_name in feature_names:
-            for suffix in reversed(feature_suffixes):
-                stop_cmds.append("sudo systemctl stop {}.{}".format(feature_name, suffix))
-                stop_cmds.append("sudo systemctl disable {}.{}".format(feature_name, suffix))
-                stop_cmds.append("sudo systemctl mask {}.{}".format(feature_name, suffix))
-            for cmd in stop_cmds:
-                syslog.syslog(syslog.LOG_INFO, "Running cmd: '{}'".format(cmd))
-                try:
-                    subprocess.check_call(cmd, shell=True)
-                except subprocess.CalledProcessError as err:
-                    syslog.syslog(syslog.LOG_ERR, "'{}' failed. RC: {}, output: {}"
-                                  .format(err.cmd, err.returncode, err.output))
-                    syslog.syslog(syslog.LOG_ERR, "Feature '{}' failed to be stopped and disabled".format(feature_name))
-                    return
-
-    def is_invariant_feature(self, feature_name, state, feature_table):
-        invariant_feature = self.cached_feature_states[feature_name] == "always_enabled" or \
-                            self.cached_feature_states[feature_name] == "always_disabled"
-        if invariant_feature:
-            invariant_state = self.cached_feature_states[feature_name]
-            if state != invariant_state:
-                syslog.syslog(syslog.LOG_INFO, "Feature '{}' service is '{}'"
-                              .format(feature_name, invariant_state))
-                entry = self.config_db.get_entry('FEATURE', feature_name)
-                entry['state'] = invariant_state
```

Comment:
@stepanblyschak Following my previous comment, if the `state` at here is `always_disabled` and `invariant_state` is `always_enabled`, the code at here will update the state field of feature to `invariant_state`. However, the code from line 761 ~ 764 will disable this feature. So I think the code at line 758 should be `entry['state'] = state`, right?

Reply:
No sure I get this comment, are you saying there is a bug in original code?
Could you please point me to a document describing feature state transitions?

Reply:
@yozhao101 Do you have this comment still?

Reply:
@yozhao101 could you please check if the last commit addresses your comment?